zone/spells.cpp

```diff
@@ -711,7 +711,13 @@ bool Mob::DoCastingChecksZoneRestrictions(bool check_on_casting, int32 spell_id)
 		Message(Chat::Red, "You cannot cast detrimental spells here.");
 		return false;
 	}
-
+	/*
+		Zones where you can not cast a spell that is for nighttime only
+	*/
+	if (spells[spell_id].time_of_day == 2 && !zone->zone_time.IsNightTime()) {
```

Comment:
Could we make an enum for this so we eliminate the magic number?

Reply:
I noticed there are some spells that uses 255.  I have no idea what that is for.      1 is day and 2 is night, although there is no spells that uses day.  I guess it's just there in case.

Reply:
Moon doesn’t have day or night if I remember right

Reply:
Luclin doesn't show time but it still uses it.  I can test this on live really quick.

Reply:
![image](https://github.com/EQEmu/Server/assets/3778417/8bc57465-1cc4-4b87-878a-d52125bab15e)

Reply:
> I noticed there are some spells that uses 255. I have no idea what that is for. 1 is day and 2 is night, although there is no spells that uses day. I guess it's just there in case.

Was just working on the assumption maybe a server operator may want that functionality for a spell eventually.

Reply:
> Yeah I could. in eq_constants?

Yes.

Reply:
updated.